### PR TITLE
[Typing][PEP585 Upgrade][BUAA][226-235] Use standard collections for type hints for 10 uts in `test/ir/inference/`

### PR DESCRIPTION
**Pull request by MufanColin**
### PR Category

User Experience

### PR Types

Devs

### Description

为如下文件添加 PEP 563，并进行 PEP 585 升级：

- ``test_trt_convert_softmax.py``
- ``test_trt_convert_solve.py``
- ``test_trt_convert_split.py``
- ``test_trt_convert_square.py``
- ``test_trt_convert_squeeze2.py``
- ``test_trt_convert_stack.py``
- ``test_trt_convert_strided_slice.py``
- ``test_trt_convert_sum.py``
- ``test_trt_convert_swish.py``
- ``test_trt_convert_take_along_axis.py``

### Related links
- #66936

@gouzil

### CODE REVIEW

File: test/ir/inference/test_trt_convert_split.py [hunk 1]
```diff
@@ -191,7 +193,7 @@ def generate_SectionsTensorList2(attrs: List[Dict[str, Any]]):
 
     def sample_predictor_configs(
         self, program_config
-    ) -> (paddle_infer.Config, List[int], float):
+    ) -> (paddle_infer.Config, list[int], float):
```
Comment (gouzil) [inline test/ir/inference/test_trt_convert_split.py]:
```suggestion
    ) -> tuple[paddle_infer.Config, list[int], float]:
```

File: test/ir/inference/test_trt_convert_split.py [hunk 2]
```diff
@@ -380,7 +382,7 @@ def generate_input1(attrs: List[Dict[str, Any]]):
...
```
Comment (gouzil) [inline test/ir/inference/test_trt_convert_split.py]:
```suggestion
    ) -> tuple[paddle_infer.Config, list[int], float]:
```

File: test/ir/inference/test_trt_convert_stack.py [hunk 3]
```diff
@@ -100,7 +102,7 @@ def generate_input3(attrs: List[Dict[str, Any]], batch):
 
     def sample_predictor_configs(
         self, program_config
-    ) -> (paddle_infer.Config, List[int], float):
+    ) -> (paddle_infer.Config, list[int], float):
```
Comment (gouzil) [inline test/ir/inference/test_trt_convert_stack.py]:
```suggestion
    ) -> tuple[paddle_infer.Config, list[int], float]:
```

File: test/ir/inference/test_trt_convert_strided_slice.py [hunk 4]
```diff
@@ -17,7 +17,10 @@
 import unittest
 from functools import partial
 from itertools import product
-from typing import Any, Generator
+from typing import TYPE_CHECKING, Any
+
+if TYPE_CHECKING:
+    from collections.abc import Generator
```
Comment (gouzil) [inline test/ir/inference/test_trt_convert_strided_slice.py]: 参考 https://github.com/PaddlePaddle/Paddle/pull/66968#discussion_r1701844792

Comment (MufanColin) [reply test/ir/inference/test_trt_convert_strided_slice.py]: 已修改

File: test/ir/inference/test_trt_convert_stack.py (hunk 3 above)
Comment (MufanColin) [reply test/ir/inference/test_trt_convert_stack.py]: 已修改

File: test/ir/inference/test_trt_convert_split.py (hunk 2 above)
Comment (MufanColin) [reply test/ir/inference/test_trt_convert_split.py]: 已修改

File: test/ir/inference/test_trt_convert_split.py (hunk 1 above)
Comment (MufanColin) [reply test/ir/inference/test_trt_convert_split.py]: 已修改

**Review by gouzil — APPROVED** [review]

**Review by SigureMo — APPROVED** [review]
LGTMeow

**Pull request merged**